### PR TITLE
Fix Crash: setMessageId:(NSString*_Nonnull)

### PR DESCRIPTION
**Pull request by anurodhp**
Got this to crash regularly with nil insertion

### CODE REVIEW

File: Monal/Classes/MLMessageProcessor.m [hunk 1]
```diff
@@ -53,6 +53,10 @@ +(void) processMessage:(XMPPMessage*) messageNode andOuterMessage:(XMPPMessage*)
         if(!errorText)
             errorText = NSLocalizedString(@"No further error description", @"");
         
+        if(!errorType) {
```
Comment (FriedrichAltheide) [inline Monal/Classes/MLMessageProcessor.m:56]: @tmolitor-stud-tu?

[the remark this answers is not in the document]
Comment (tmolitor-stud-tu) [reply Monal/Classes/MLMessageProcessor.m:56]: would be nice to see the xml that is triggering that crash

[the remark this answers is not in the document]
Comment (anurodhp) [reply Monal/Classes/MLMessageProcessor.m:56]: I think it was your server actually

[the remark this answers is not in the document]
Comment (anurodhp) [reply Monal/Classes/MLMessageProcessor.m:56]: Regardless a malformed stanza or buggy server shouldn’t be able to crash the app

[the remark this answers is not in the document]
Comment (anurodhp) [reply Monal/Classes/MLMessageProcessor.m:56]: @tmolitor-stud-tu 
this causes the crash. There are two issues one is that this error type isnt  caught the other is we really do not want the app crashing regardless

<message xml:lang='en' id='D3B7C67B-B495-4A14-9901-FA322DDEFDDC' xmlns='jabber:client' type='error' from='anu@yax.im/converse.js-122012939'><origin-id id='D3B7C67B-B495-4A14-9901-FA322DDEFDDC' xmlns='urn:xmpp:sid:0'/><received id='0d45949e-1d2f-4110-8a3d-4e74f1a9fce5' xmlns='urn:xmpp:receipts'/><store xmlns='urn:xmpp:hints'/><stanza-id id='96e10801-e9d8-471a-b907-3098d5c83d15' xmlns='urn:xmpp:sid:0' by='anu@yax.im'/><error><recipient-unavailable/></error><delay stamp='2021-02-07T04:48:23Z' xmlns='urn:xmpp:delay'/></message>

[the remark this answers is not in the document]
Comment (FriedrichAltheide) [reply Monal/Classes/MLMessageProcessor.m:56]: @tmolitor-stud-tu I think we should allow nil in the db function instead
-(void) setMessageId:errorType:errorReason

[the remark this answers is not in the document]
Comment (tmolitor-stud-tu) [reply Monal/Classes/MLMessageProcessor.m:56]: interesting xml...

we can't fix the app against missbehaving servers, checking all input from the server would easily add a lot of checking code that makes our codebase more unreadable and add little to no benefit: usually you have to trust your own server to be sane and that is okay, server bugs are usually very rare...

@FriedrichAltheide I don't think so, the fix anu did is right for this piece of xml

[the remark this answers is not in the document]
Comment (tmolitor-stud-tu) [reply Monal/Classes/MLMessageProcessor.m:56]: I analyzed this stanza with georg and his server logs show that this seems to be a bug in converse.js

**Comment by anurodhp** [comment]
A converse bug actually makes it worse as another client can crash monal :)

**Comment by tmolitor-stud-tu** [comment]
I know :D
But Georg and I could not find this error in the code of conversejs

Am 8. Februar 2021 19:41:59 MEZ schrieb Anu Pokharel <notifications@github.com>:

**Pull request merged**